packages/react-components/src/components/TypingIndicator.tsx

```diff
@@ -56,57 +147,79 @@ const getDefaultComponents = (
     if (totalCharacterCount + additionalCharCount <= MAXIMUM_LENGTH_OF_TYPING_USERS || countOfUsersMentioned === 1) {
-      typingUsersMentioned.push(typingUser);
+      if (typingUser.displayName) {
+        typingUsersMentioned.push(typingUser);
+      }
       totalCharacterCount += additionalCharCount;
       countOfUsersMentioned += 1;
     } else {
       break;
     }
   }
 
-  typingUsersMentioned.forEach((typingUser, index) => {
-    displayComponents.push(
-      <span
-        className={mergeStyles(typingIndicatorListStyle, styles?.typingUserDisplayName)}
-        key={`typing indicator display string ${index}`}
-      >
-        {index < typingUsers.length - 1 ? typingUser.displayName + ', ' : typingUser.displayName}
-      </span>
-    );
-  });
-
   const countOfUsersNotMentioned = typingUsers.length - typingUsersMentioned.length;
-  if (countOfUsersNotMentioned > 0) {
-    displayComponents.push(
-      <span className={mergeStyles(typingIndicatorVerbStyle, styles?.typingString)} key={'UsersCountKey'}>
-        {` and ${countOfUsersNotMentioned} other${countOfUsersNotMentioned === 1 ? '' : 's'}`}
-      </span>
-    );
-  }
 
-  return displayComponents;
-};
+  const userElements: JSX.Element[] = [];
+  typingUsersMentioned.forEach((user, index) => {
+    userElements.push(onRenderUser ? onRenderUser(user) : <span key={`user-${index}`}>{user.displayName}</span>);
+    userElements.push(<span key={`comma-${index}`}>{`${strings.comma} `}</span>);
+  });
+  // pop last comma
+  userElements.pop();
+  const usersElement = (
+    <Stack horizontal className={mergeStyles(typingIndicatorStringStyle, styles?.typingUserDisplayName)}>
+      {userElements}
+    </Stack>
+  );
+  let variables: Record<string, JSX.Element> = {};
+  let typingString = '';
+  if (typingUsers.length === 1) {
+    typingString = strings.singleUser;
+    variables = {
+      user: usersElement
+    };
+  } else if (typingUsers.length > 1 && countOfUsersNotMentioned === 0) {
+    typingString = strings.multipleUsers;
+    variables = {
+      users: usersElement
+    };
+  } else if (typingUsers.length > 1 && countOfUsersNotMentioned === 1) {
+    typingString = strings.multipleUsersAbbreviateOne;
+    variables = {
+      users: usersElement
+    };
+  } else if (typingUsers.length > 1 && countOfUsersNotMentioned > 1) {
+    typingString = strings.multipleUsersAbbreviateMany;
+    variables = {
+      users: usersElement,
+      numOthers: <>{countOfUsersNotMentioned}</>
+    };
+  }
 
-const defaultTypingString = (typingUsers: CommunicationParticipant[]): string => {
-  return typingUsers.length > 0 ? (typingUsers.length > 1 ? ' are typing...' : ' is typing...') : '';
+  return (
+    <Stack horizontal className={mergeStyles(typingIndicatorStringStyle, styles?.typingString)} key="typingStringKey">
+      {formatElements(typingString, variables)}
+    </Stack>
+  );
```

Comment:
nit: this function is ~70 lines long, considering breaking up into multiple functions